apps/portals/b2ai.standards/src/components/ChallengesCardDeck.tsx

```diff
@@ -1,66 +1,183 @@
-      columnName: DST_TABLE_COLUMN_NAMES.RELEVANT_ORG_NAMES,
-      facetValues: ['Precision Public Health Grand Challenge'],
-    },
-  ],
+type ObjList = {
+  [key: string]: string
+}[]
+type useTableFetchProps = {
+  entityId: string
+  columns: string[]
+  sql?: string
+  shouldRun?: boolean
 }
 
-const CLINICAL_CARE_CHALLENGE_FILTERED_QUERY: Query = {
-  sql: dataSql,
-  limit: 25,
-  selectedFacets: [
-    {
-      concreteType:
-        'org.sagebionetworks.repo.model.table.FacetColumnValuesRequest',
-      columnName: DST_TABLE_COLUMN_NAMES.RELEVANT_ORG_NAMES,
-      facetValues: ['AI/ML for Clinical Care Grand Challenge'],
-    },
-  ],
-}
+export function useTableFetch({
+  entityId,
+  columns, // required
+  sql, // if filters or aliases or anything are needed, pass in whole sql query
+  shouldRun = true, // if calling before ready, send false
+}: useTableFetchProps) {
+  let data: ObjList = []
+  const error = undefined
 
-const CHALLENGE_CARDS: CardDeckCardProps[] = [
-  {
-    title: 'Precision Public Health Challenge',
-    description:
-      'Developing an ethically-sourced and diverse voice database linked to multimodal health biomarkers to assist in screening, diagnosis, and treatment of diseases.',
-    ctaButtonText: 'Explore Standards',
-    ctaButtonURL: createExplorePageLink(
-      PRECISION_PUBLIC_HEALTH_CHALLENGE_FILTERED_QUERY,
-    ),
-    headerImage: <img src={soundSvg} />,
-  },
-  {
-    title: 'Clinical Care Challenge',
-    description:
-      'Establishing the most diverse, high-resolution, ethically sourced, AI-ready data set to explore questions around improving recovery from acute illness',
-    ctaButtonText: 'Explore Standards',
-    ctaButtonURL: createExplorePageLink(CLINICAL_CARE_CHALLENGE_FILTERED_QUERY),
-    headerImage: <img src={healthSvg} />,
-  },
-]
+  if (!sql) {
+    sql = `SELECT ${columns.join(', ')} FROM ${entityId}`
+  }
+
+  const queryBundleRequest: QueryBundleRequest = {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    partMask:
+      SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS |
+      SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    query: { sql },
+  }
+  const queryResultBundle = useGetQueryResultBundle(queryBundleRequest, {
+    enabled: shouldRun,
+  })
+
+  if (!shouldRun || error || !queryResultBundle?.data) {
+    return { data, error }
+  }
+
+  const colIndexes = columns.map(column => ({
+    column,
+    index: getFieldIndex(column, queryResultBundle.data),
+  }))
+
+  data =
+    queryResultBundle?.data?.queryResult?.queryResults.rows.map(el => {
+      const values = el.values as string[]
+      if (values.some(value => value === null)) {
+        console.warn('Row has null value(s) when no nulls expected')
+      }
+      const result = {}
+      colIndexes.forEach(({ column, index }) => {
+        result[column] = values[index]
+      })
+      return result
+    }) ?? []
+
+  return { data, error }
+}
```

Comment:
Why not use [`select`](https://tanstack.com/query/latest/docs/framework/react/guides/render-optimizations#select) to transform the data, then `useTableFetch` will return a bunch of useful things, like `isLoading`? Instead of dropping everything except `data` and `error`

```suggestion
  return useGetQueryResultBundle(queryBundleRequest, {
    enabled: shouldRun,
    select: (data) => {
        const colIndexes = columns.map(column => ({
          column,
          index: getFieldIndex(column, queryResultBundle.data),
        }))
        return data?.queryResult?.queryResults.rows.map(el => {
          const values = el.values as string[]
          if (values.some(value => value === null)) {
            console.warn('Row has null value(s) when no nulls expected')
          }
          const result = {}
          colIndexes.forEach(({ column, index }) => {
            result[column] = values[index]
          })
          return result
        }) ?? []
    }
  })
}
```